### PR TITLE
implemented youtube channel table

### LINKED ISSUE #7580: [YouTube] Implement the Channel Resource table to contain info about a YouTube channel
MindsDB integrates with YouTube, so you can access YouTube data and make relevant predictions with MindsDB.

## Resources

Familiarize yourself with the existing YouTube integration: https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/youtube_handler

## Steps to Follow

Currently, the YouTube integration with MindsDB accesses the comments table.

The task is to extend [this implementation](https://github.com/mindsdb/mindsdb/tree/staging/mindsdb/integrations/handlers/youtube_handler) to include the Channel Resource table to contain info about a YouTube channel.

## :loudspeaker: Additional rewards :trophy:

Each integration extension issue brings 15+ points for SWAG and entry into the draw for a :point_down: :
-  :computer:  [Razer Blade 16 Laptop](https://www.razer.com/gaming-laptops/Razer-Blade-16/RZ09-0483SEJ3-R3U1)
- :shirt: :bear:  [MindsDB Swag](https://store.covver.io/mindsdb/collections/swag-collection)

:information_source: For more info check out https://mindsdb.com/hacktoberfest/ :point_left:

### PR DESCRIPTION
**Pull request by Better-Boy**
## Description

implemented youtube channel table

Fixes #7580 

## Type of change

(Please delete options that are not relevant)

- [ ] 🐛 Bug fix (non-breaking change which fixes an issue)
- [x] ⚡ New feature (non-breaking change which adds functionality)
- [ ] 📢 Breaking change (fix or feature that would cause existing functionality not to work as expected)
- [ ] 📄 This change requires a documentation update

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

Screenshot 2023-10-05 at 11 16 15 PM

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [ ] Necessary documentation updates are either made or tracked in issues.
- [ ] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by martyna-mindsdb** [comment]
@Better-Boy 
Please update the youtube_handler/README.md file.

**Comment by martyna-mindsdb** [comment]
@Better-Boy 
Currently, the WHERE clause requires channel_id to be defined. Can you implement an alternative for it? Namely, let's allow the WHERE clause to have channel_name defined (instead or along with channel_id).

**Comment by Better-Boy** [comment]
> @Better-Boy Please update the youtube_handler/README.md file.

Done

File: mindsdb/integrations/handlers/youtube_handler/youtube_tables.py [hunk 1]
```diff
@@ -148,3 +145,59 @@ def call_youtube_comments_api(self,video_id):
...
+        channel_id = None
+        for op, arg1, arg2 in where_conditions:
+            if arg1 == 'channel_id':
+                if op == '=':
+                    channel_id = arg2
+                    break
+                else:
+                    raise NotImplementedError("Only '=' operator is supported for channel_id column.")
+
+        if not channel_id:
```
Comment (MinuraPunchihewa) [inline mindsdb/integrations/handlers/youtube_handler/youtube_tables.py:172]: Same question as before with your other PR, is it always mandatory to include the channel ID?

Comment (MinuraPunchihewa) [reply mindsdb/integrations/handlers/youtube_handler/youtube_tables.py:172]: If it absolutely is, please make sure you mention this in the README.

**Review by MinuraPunchihewa — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/youtube_handler/youtube_tables.py (hunk 1 above)
Comment (Better-Boy) [reply mindsdb/integrations/handlers/youtube_handler/youtube_tables.py:172]: Channel ID is needed. Updated readme.

**Review by MinuraPunchihewa — APPROVED** [review]

**Pull request merged**